pkg/apis/duck/v1alpha1/annotations_test.go

```diff
@@ -208,3 +210,68 @@ func TestValidateAutoscalingAnnotations(t *testing.T) {
 		})
 	}
 }
+
+func TestSetClusterNameAnnotation(t *testing.T) {
+	testCases := map[string]struct {
+		orig     *v1.ObjectMeta
+		expected *v1.ObjectMeta
+	}{
+		"no annotation": {
+			orig: &v1.ObjectMeta{},
+			expected: &v1.ObjectMeta{
+				Annotations: map[string]string{
+					ClusterNameAnnotation: testingMetadataClient.FakeClusterName,
+				},
+			},
+		},
+		"has annotation": {
+			orig: &v1.ObjectMeta{
+				Annotations: map[string]string{
+					ClusterNameAnnotation: "testing-cluster-name",
+				},
+			},
+			expected: &v1.ObjectMeta{
+				Annotations: map[string]string{
+					ClusterNameAnnotation: "testing-cluster-name",
+				},
+			},
+		},
+	}
+	for n, tc := range testCases {
+		t.Run(n, func(t *testing.T) {
+			SetClusterNameAnnotation(tc.orig, testingMetadataClient.NewTestClient())
+			if diff := cmp.Diff(tc.expected, tc.orig); diff != "" {
+				t.Errorf("Unexpected differences (-want +got): %v", diff)
+			}
+		})
+	}
+}
+
+func TestValidateClusterNameAnnotation(t *testing.T) {
+	testCases := map[string]struct {
+		objMeta *v1.ObjectMeta
+		error   bool
```

Comment:
Is there any way to test an error case for this method? Does the metadata testing client support mocking errors?